unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/SlitPlate.cs

```diff
@@ -0,0 +1,302 @@
+                            }
+                        }
+                    }                             
+                }
+            }
+            else
+            {
+                DeactivatePlateWaveGenerators();
+            }
+        }
+
+        private int CalculateGeneratorsPerSlit()
+        {        
+            int numberOfGeneratorsPerSlit = 1 + (int)((slitWidth * 10.0f)/1.5f);
+            return numberOfGeneratorsPerSlit; 
+        }
+
+        public void AddWaveGenerator()
+        {
+            var waveGenerator = new GameObject("WaveGenerator");
+            waveGenerator.transform.parent = GameObject.Find("Plate").transform;
+            waveGenerator.transform.rotation = waveGenerator.transform.parent.rotation;
+            //waveGenerator.transform.localPosition = left.transform.localPosition;
+
+            WaveGenerator waveGeneratorScript = waveGenerator.gameObject.AddComponent<WaveGenerator>();
+            waveGeneratorScript.WaveAmplitude = 0.5f;
+            waveGeneratorScript.WaveLength = 0.25f;
+            waveGeneratorScript.WaveFrequency = 0.5f;
+            waveGeneratorScript.SetPropagationMode(WaveGenerator.WavePropagation.Circular);
+            if (GameObject.Find("Water") != null)
+            {
+                waveGeneratorScript.SetPlaneObject(GameObject.Find("Water"));
+                waveGeneratorList.Add(waveGenerator);
+            }
+        }
+
+        public void AddAllWaveGenerators() 
+        {          
+            var totalNumberOfGenerators = generatorCountPerSlit * numberOfSlits; 
+            for (int count = 0; count < totalNumberOfGenerators - 1; count++)
+            {
+                AddWaveGenerator();
+            }
+            RegisterPlateWaveGenerators();
+        }
+
+        private void RegisterPlateWaveGenerators()
+        {
+            foreach (var generator in waveGeneratorList)
+                if(generator != waveGenerator)
+                    waterPlane.RegisterWaveGenerator(generator.GetComponent<WaveGenerator>());
+        }
+
+        private void ActivatePlateWaveGenerators()
+        {
+            foreach (var generator in waveGeneratorList)
+                generator.GetComponent<WaveGenerator>().SetGeneratorActive(true);
+        }
+
+        private void DeactivatePlateWaveGenerators()
+        {
+            foreach (var generator in waveGeneratorList)
+                generator.GetComponent<WaveGenerator>().SetGeneratorActive(false);
+        }
+
+        public void ResetObject()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void ResetCubes()
+        {
+            foreach (var section in midSections){
+                DestroyImmediate(section);   
+            }
+
+            ResetWaveGenerators();
+            midSections.Clear();             
+        }
+
+        public void ResetWaveGenerators()
+        {
+            foreach (var gen in waveGeneratorList)

```

Comment:
We can simplify this to:
foreach (var gen in waveGeneratorList.Where(gen => gen != waveGenerator).Where(gen => gen != null))